Build failure: ike-scan

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

1. Add ike-scan to environment.systemPackages list.
2. nixos-rebuild switch

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/297566824

### Relevant log output

```console
configure: error: cannot determine 64-bit integer type
```

### Additional context

The only change is the update to nixos 25.05. The package version is the same.

### System metadata

- system: `"x86_64-linux"`
- host os: `Linux 6.12.30, NixOS, 25.05 (Warbler), 25.05.20250528.7282cb5`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.28.3`
- nixpkgs: `/nix/store/01635adignbizr4nyz5lpawkrxn4n4g8-source`

### Notify maintainers

@fabaff

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ik/ike-scan/package.nix
+++ b/pkgs/by-name/ik/ike-scan/package.nix
@@ -9,14 +9,23 @@
 
 stdenv.mkDerivation rec {
   pname = "ike-scan";
-  version = "1.9.5";
+  version = "1.9.5-unstable-2024-09-15";
 
   src = fetchFromGitHub {
     owner = "royhills";
     repo = "ike-scan";
-    rev = version;
-    sha256 = "sha256-mbfg8p3y4aKoXpmLuF9GXAMPEqV5CsvetwGCRDJ9UNY=";
+    rev = "c74c01fd22d9a3aae3d8ba9a0bd2eb1a2146ac6f";
+    hash = "sha256-+eicvirqzZrAJiaGaVjqZlSpU2+jTG/MRPv50P+1Tpc=";
   };
+
+  patches = [
+    # Using the same patches as for the Fedora RPM
+    (fetchpatch {
+      # Memory leaks, https://github.com/royhills/ike-scan/pull/15
+      url = "https://github.com/royhills/ike-scan/pull/15/commits/d864811de08dcddd65ac9b8d0f2acf5d7ddb9dea.patch";
+      hash = "sha256-VVJZSTZfDV0qHuxdNoZV1NXJYCEMtB0bO1oi2hLCeXE=";
+    })
+  ];
 
   nativeBuildInputs = [
     autoreconfHook
@@ -25,15 +34,6 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "--with-openssl=${openssl.dev}" ];
 
-  patches = [
-    # Using the same patches as for the Fedora RPM
-    (fetchpatch {
-      # Memory leaks, https://github.com/royhills/ike-scan/pull/15
-      url = "https://github.com/royhills/ike-scan/pull/15/commits/d864811de08dcddd65ac9b8d0f2acf5d7ddb9dea.patch";
-      sha256 = "0wbrq89dl8js7cdivd0c45hckmflan33cpgc3qm5s3az6r4mjljm";
-    })
-  ];
-
   meta = with lib; {
     description = "Tool to discover, fingerprint and test IPsec VPN servers";
     longDescription = ''
@@ -41,7 +41,7 @@ stdenv.mkDerivation rec {
       fingerprint and test IPsec VPN servers.
     '';
     homepage = "https://github.com/royhills/ike-scan";
-    license = with licenses; [ gpl3Plus ];
+    license = licenses.gpl3Plus;
     platforms = platforms.linux;
     maintainers = with maintainers; [ fab ];
   };

